server/elasticsearch/src/main/java/org/hawkular/apm/server/elasticsearch/SpanServiceElasticsearch.java

```diff
@@ -91,6 +92,16 @@ public Span getSpan(String tenantId, String id) {
         if (!response.isSourceEmpty()) {
             try {
                 span = deserialize(response.getSourceAsString(), Span.class);
+
+                /**
+                 * Enrich server span with client span annotations
+                 */
+                if (span.serverSpan() && span.url() == null) {
+                    Span clientSpan = getSpan(tenantId, SpanUniqueIdGenerator.getClientId(span.getId()));
+                    List<BinaryAnnotation> binaryAnnotations = new ArrayList<>(clientSpan.getBinaryAnnotations());
```

Comment:
What if clientSpan is null? (i.e. the client span is not found)

Copying all binary annotations from the client span may be an issue, as could be misleading for the resulting server span. However only alternative may be to 'set' the URL on the server span (i.e. override any value that it potentially derived).